### PR TITLE
Require another approval to increase priority/throughput beyond limits

### PR DESCRIPTION
**Pull request by peregrineshahin**
Supersedes https://github.com/official-stockfish/fishtest/pull/1830
Dependent on  https://github.com/official-stockfish/fishtest/pull/1934

### CODE REVIEW

File: server/fishtest/views.py
Comment (XInTheDark) [inline server/fishtest/views.py:1224]: `!=` should probably be changed to `<` here, since we can allow e.g. decreasing throughput if it was set to above 100 previously

Comment (peregrineshahin) [reply server/fishtest/views.py:1224]: > `!=` should probably be changed to `<` here, since we can allow e.g. decreasing throughput if it was set to above 100 previously

Please look at the code again, != here is only the first condition to check if before != new i.e. check if changed

**Comment by peregrineshahin** [comment]
Indeed that was the intended message.

**Comment by peregrineshahin** [comment]
Hmm thinking about it probably it would make better sense if we allow the user to change the prio but at the same time put the test again in the pending section waiting for approval?

**Comment by vdbergh** [comment]
> > DEV updated. Even when the run transitions to the approval state, the workers should continue to post results. We need to investigate whether this is causing any disruptions to the run state machine, which is currently undocumented.
> 
> What happens currently if a modified worker decided to submit a result for non approved test?

I think he can only submit results if he obtains a task. And I assume he will only be able to obtain a task for an approved test. But it would not matter to the worker if the test were suddenly unapproved (it does not care about the server state).

**Comment by peregrineshahin** [comment]
Hmmm, so you are saying that if the test was unapproved suddenly the workers will still send the result of their tasks and the results will be accepted for the older tasks before approving, but it will not get new tasks after all the after the unapproval proccess?
I'm okay with that if that's the case.
i.e. no need to ditch the results/the tasks.

**Comment by vdbergh** [comment]
Yes that’s what I think is true.

File: server/fishtest/views.py
Comment (vdbergh) [inline server/fishtest/views.py:1233]: Perhaps flash message here?

**Comment by vdbergh** [comment]
So with this PR it seems an finished run can be unapproved.... So no rule in the schema of the form :smiling_face_with_tear: 
```python
ifthen(keys("finished"), keys("approved"))
```

**Comment by ppigazzini** [comment]
DEV updated.

**Review by ppigazzini — APPROVED** [review]
Looks good on DEV

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin

**Pull request merged**
